Applications/ConsumerApi/src/Controllers/Onboarding/OnboardingController.cs

```diff
@@ -0,0 +1,106 @@
+    {
+        var pickedOnboardingConfiguration = PickAppSpecificConfiguration(appName);
+
+        if (pickedOnboardingConfiguration != null)
+        {
+            var appStoreList = new List<Tuple<string, string>>();
+            var userAgentOfRequest = Request.Headers["User-Agent"].ToString();
+
+            // Note: this order is important as iPhoneRequest will match with MacOsRequest also.
+            if (IndicatesIPhone(userAgentOfRequest))
+                appStoreList.Add(new Tuple<string, string>("Apple App Store", AppendDeviceHint(pickedOnboardingConfiguration.IosAppUrl, I_PHONE_DEVICE_HINT)));
+            else if (IndicatesMacOs(userAgentOfRequest))
+                appStoreList.Add(new Tuple<string, string>("Apple App Store", AppendDeviceHint(pickedOnboardingConfiguration.IosAppUrl, MAC_OS_DEVICE_HINT)));
+
+            if (IndicatesAndroid(userAgentOfRequest))
+                appStoreList.Add(new Tuple<string, string>("Google Play Store", pickedOnboardingConfiguration.AndroidAppUrl));
+
+            if (appStoreList.Count == 0)
+            {
+                appStoreList.Add(new Tuple<string, string>("Google Play Store", pickedOnboardingConfiguration.AndroidAppUrl));
+                appStoreList.Add(new Tuple<string, string>("Apple App Store", pickedOnboardingConfiguration.IosAppUrl));
+            }
+
+            if (string.IsNullOrEmpty(Request.Path.Value)) return BadRequest("Invalid request path");
+
+            return _onboardingHtmlFactory.GenerateOnboardingPage(pickedOnboardingConfiguration, appStoreList);
+        }
+
+        return _onboardingHtmlFactory.GenerateAppSelectionPage(_configuration, Request);
+    }
+
+    private ConsumerApiConfiguration.OnboardingConfiguration? PickAppSpecificConfiguration(string? appname)
+    {
+        var onboardingConfigurations = _configuration.Onboarding;
+
+        // the length can't be null, as it is required by the configuration
+
+        if (onboardingConfigurations.Length == 1)
+            return onboardingConfigurations[0];
+
+        foreach (var appConfig in onboardingConfigurations)
+            if (appConfig.AppNameIdentifier.Equals(appname))
+                return appConfig;
+
+        return null;
+    }
+
+    private bool IndicatesAndroid(string userAgentContent)
+    {
+        return userAgentContent.ToLower().Contains(ANDROID_IDENTIFIER);
+    }
+
+    private bool IndicatesMacOs(string userAgentContent)
+    {
+        if (userAgentContent.ToLower().Contains(IOS_IDENTIFIER))
+            return true;
+        if (userAgentContent.ToLower().Contains(MACINTOSH_IDENTIFIER))
+            return true;
+
+        return false;
+    }
+
+    private bool IndicatesIPhone(string userAgentContent)
+    {
+        return userAgentContent.ToLower().Contains(IPHONE_IDENTIFIER);
+    }
+
+    private string AppendDeviceHint(string url, string deviceHint)
+    {
+        return url.Contains("?") ? $"{url}&platform={deviceHint}" : $"{url}?platform={deviceHint}";
```

Comment:
I think you should use some library (or maybe there is something built into .NET) to add the query parameter. Because:
1. it currently looks ugly. :)
2. there might be cases you don't think about right now that your code can't handle.